src/main/java/org/mockito/internal/util/reflection/GenericMetadataSupport.java

```diff
@@ -62,6 +62,42 @@
      */
     protected Map<TypeVariable, Type> contextualActualTypeParameters = new HashMap<TypeVariable, Type>();
 
+    /**
+     * Registers the type variables for the given type and all of its superclasses and superinterfaces.
+     */
+    protected void registerAllTypeVariables(Type classType) {
+        if (!(classType instanceof Class || classType instanceof ParameterizedType)) {
+            return;
+        }
+
+        Queue<Type> typesToRegister = new LinkedList<Type>();
+        Set<Type> registeredTypes = new HashSet<Type>();
+        typesToRegister.add(classType);
+
+        while (!typesToRegister.isEmpty()) {
+            Type typeToRegister = typesToRegister.poll();
+            if (typeToRegister == null || registeredTypes.contains(typeToRegister)) {
+                continue;
+            }
+
+            registerTypeVariablesOn(typeToRegister);
+            registeredTypes.add(typeToRegister);
+
+            Class<?> rawType = getRawType(typeToRegister);
+            typesToRegister.add(rawType.getGenericSuperclass());
+            typesToRegister.addAll(Arrays.asList(rawType.getGenericInterfaces()));
+        }
+    }
+
+    private Class<?> getRawType(Type clazz) {
+        if (clazz instanceof ParameterizedType) {
+            return getRawType(((ParameterizedType) clazz).getRawType());
+        } else if (clazz instanceof Class) {
+            return (Class<?>) clazz;
+        } else {
```

Comment:
This else clause is always when `clazz instanceof Class` so this Exception is not required. See above comment.